cli/lsp/completions.rs

```diff
@@ -0,0 +1,771 @@
+    node: &swc_ast::ImportDecl,
+    _parent: &dyn Node,
+  ) {
+    if self.maybe_specifier.is_none() {
+      let start = self.source_map.lookup_char_pos(node.src.span.lo);
+      let end = self.source_map.lookup_char_pos(node.src.span.hi);
+      if span_includes_pos(&self.position, &start, &end) {
+        self.maybe_range = Some(lsp::Range {
+          start: lsp::Position {
+            line: (start.line - 1) as u32,
+            character: (start.col_display + 1) as u32,
+          },
+          end: lsp::Position {
+            line: (end.line - 1) as u32,
+            character: (end.col_display - 1) as u32,
+          },
+        });
+        self.maybe_specifier = Some(node.src.value.to_string());
+      }
+    }
+  }
+
+  fn visit_named_export(
+    &mut self,
+    node: &swc_ast::NamedExport,
+    _parent: &dyn Node,
+  ) {
+    if self.maybe_specifier.is_none() {
+      if let Some(src) = &node.src {
+        let start = self.source_map.lookup_char_pos(src.span.lo);
+        let end = self.source_map.lookup_char_pos(src.span.hi);
+        if span_includes_pos(&self.position, &start, &end) {
+          self.maybe_specifier = Some(src.value.to_string());
+        }
+      }
+    }
+  }
+
+  fn visit_export_all(
+    &mut self,
+    node: &swc_ast::ExportAll,
+    _parent: &dyn Node,
+  ) {
+    if self.maybe_specifier.is_none() {
+      let start = self.source_map.lookup_char_pos(node.src.span.lo);
+      let end = self.source_map.lookup_char_pos(node.src.span.hi);
+      if span_includes_pos(&self.position, &start, &end) {
+        self.maybe_specifier = Some(node.src.value.to_string());
+      }
+    }
+  }
+
+  fn visit_ts_import_type(
+    &mut self,
+    node: &swc_ast::TsImportType,
+    _parent: &dyn Node,
+  ) {
+    if self.maybe_specifier.is_none() {
+      let start = self.source_map.lookup_char_pos(node.arg.span.lo);
+      let end = self.source_map.lookup_char_pos(node.arg.span.hi);
+      if span_includes_pos(&self.position, &start, &end) {
+        self.maybe_specifier = Some(node.arg.value.to_string());
+      }
+    }
+  }
+}
+
+/// Determine if the provided position falls into an module specifier of an
+/// import/export statement, optionally returning the current value of the
+/// specifier.
+fn is_module_specifier_position(
+  specifier: &ModuleSpecifier,
+  source: &str,
+  media_type: &MediaType,
+  position: &lsp::Position,
+) -> Option<(String, lsp::Range)> {
+  if let Ok(parsed_module) =
+    analysis::parse_module(specifier, source, media_type)
+  {
+    let mut import_locator =
+      ImportLocator::new(*position, parsed_module.source_map.clone());
+    parsed_module
+      .module
+      .visit_with(&swc_ast::Invalid { span: DUMMY_SP }, &mut import_locator);
+    if let (Some(specifier), Some(range)) =
+      (import_locator.maybe_specifier, import_locator.maybe_range)
+    {
+      Some((specifier, range))
+    } else {
+      None
+    }
+  } else {
+    None
+  }
+}
+
+fn relative_specifier(
```

Comment:
Could you add a comment to describe what this function does? It is not immediately obvious to me from looking at signature / tests.